src/transformers/modeling_tf_utils.py

```diff
@@ -300,11 +304,13 @@ def hf_compute_loss(self, labels, logits):
         )
         # make sure only labels that are not equal to -100
         # are taken into account as loss
-        next_sentence_active_loss = tf.not_equal(tf.reshape(labels, (-1,)), -100)
-        next_sentence_reduced_logits = tf.boolean_mask(tf.reshape(logits, (-1, 2)), next_sentence_active_loss)
-        next_sentence_label = tf.boolean_mask(tf.reshape(labels, (-1,)), next_sentence_active_loss)
 
-        return loss_fn(next_sentence_label, next_sentence_reduced_logits)
+        unmasked_ns_loss = loss_fn(y_true=labels, y_pred=logits)
+        ns_loss_mask = tf.cast(labels != -100, dtype=unmasked_ns_loss.dtype)
+        # Just zero out samples where label is -100, no reduction
+        masked_ns_loss = tf.math.multiply_no_nan(unmasked_ns_loss, ns_loss_mask)
+
+        return masked_ns_loss
```

Comment:
Should this one be reduced as well?

Reply:
`ns_loss` is calculated per sample rather than per position, so `masked_ns_loss` is a vector of shape `(num_samples,)`. We could reduce that to a scalar if we want, though!